x/evm/keeper/abci.go

```diff
@@ -16,10 +17,29 @@ import (
 
 // BeginBlock sets the sdk Context and EIP155 chain id to the Keeper.
 func (k *Keeper) BeginBlock(ctx sdk.Context, _ abci.RequestBeginBlock) {
+	logger := ctx.Logger().With("begin_block", "evm")
 	k.WithChainID(ctx)
 	// check for eth_tx_log events at BeginBlock
 	// and update block bloom filter with them (if any)
-	k.updateBlockBloom(ctx)
+	k.updateBlockBloom(ctx, logger)
+
+	// Base fee is already set on FeeMarket BeginBlock
+	// that runs before this one
+	// We emit this event on the EVM and FeeMarket modules
+	// because they can be different if the evm denom has 6 decimals
+	res, err := k.BaseFee(ctx, &types.QueryBaseFeeRequest{})
+	if err != nil {
+		logger.Error("error when getting base fee", "error", err.Error())
+	}
+	if res != nil && res.BaseFee != nil {
+		// Store current base fee in event
+		ctx.EventManager().EmitEvents(sdk.Events{
+			sdk.NewEvent(
+				types.EventTypeFeeMarket,
+				sdk.NewAttribute(types.AttributeKeyBaseFee, res.BaseFee.String()),
+			),
+		})
```

Comment:
Can't we deprecate the event emitted in the FeeMarket if this is the one with the correct decimals?

Reply:
Not sure if that will break any current application.
We need this new event because on historical queries via JSON-RPC we get the base fee from the event. So this event will always be 18 decimals, because is within the evm module.
The event with the correct decimals (the IBC coin decimals) is the one from the feemarket